ocaml/typing/typemod.ml

```diff
@@ -100,10 +100,25 @@ type error =
       old_source_file : Misc.filepath;
     }
   | Submode_failed of Mode.Value.error
+  | Underscore_not_allowed_in_signature
+  | Cannot_infer_module_type
 
 exception Error of Location.t * Env.t * error
 exception Error_forward of Location.error
 
+(** When calling [transl_modtype_decl], there are three possibilities for its
+    [expected_modtype_decl]:
+    - [From_signature]: we are checking a signature.
+    - [No_expected]: we are checking a struct, but we don't have information about
+       its expected signature.
+    - [Expected _]: we are checking a struct, and we do have information about its
+       expected signature (where "expected signature" is used for inferring module types
+       when we see a [module type S = _]). *)
+type expected_modtype_decl =
+  | From_signature
+  | No_expected
+  | Expected of modtype_declaration
```

Comment:
Might be clearer to refactoring this into:
```
type modtype_decl_context =
   | In_signature
   | In_structure of modtype_declaration option
```
Because in the case of `In_signature` it's not much of an "expected modtype_decl".

Reply:
Also a minor formatting nit: might be clearer if you put the comments next to each constructor. (with `(** *)` which will be recognized as doc generator)